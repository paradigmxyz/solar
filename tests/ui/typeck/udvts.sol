type U01 is U01;     //~ ERROR: the underlying type of UDVTs must be an elementary value type
type U02 is string;  //~ ERROR: the underlying type of UDVTs must be an elementary value type
type U03 is bytes;   //~ ERROR: the underlying type of UDVTs must be an elementary value type
type U04 is uint[];  //~ ERROR: the underlying type of UDVTs must be an elementary value type
type U05 is U06;     //~ ERROR: the underlying type of UDVTs must be an elementary value type

type U06 is address;
type U07 is address payable;
type U08 is uint;
type U09 is uint256;
type U10 is int;
type U11 is int256;
