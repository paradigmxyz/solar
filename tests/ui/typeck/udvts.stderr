error: the underlying type of UDVTs must be an elementary value type
  --> ROOT/tests/ui/typeck/udvts.sol:LL:CC
   |
LL | type U01 is U01;
   |             ^^^
   |

error: the underlying type of UDVTs must be an elementary value type
  --> ROOT/tests/ui/typeck/udvts.sol:LL:CC
   |
LL | type U02 is string;
   |             ^^^^^^
   |

error: the underlying type of UDVTs must be an elementary value type
  --> ROOT/tests/ui/typeck/udvts.sol:LL:CC
   |
LL | type U03 is bytes;
   |             ^^^^^
   |

error: the underlying type of UDVTs must be an elementary value type
  --> ROOT/tests/ui/typeck/udvts.sol:LL:CC
   |
LL | type U04 is uint[];
   |             ^^^^^^
   |

error: the underlying type of UDVTs must be an elementary value type
  --> ROOT/tests/ui/typeck/udvts.sol:LL:CC
   |
LL | type U05 is U06;
   |             ^^^
   |

error: aborting due to 5 previous errors

